libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/barrier.h

```diff
@@ -561,6 +553,68 @@ inline _CUDA_VSTD::uint64_t * barrier_native_handle(barrier<thread_scope_block>
     return reinterpret_cast<_CUDA_VSTD::uint64_t *>(&b.__barrier);
 }
 
+
+_LIBCUDACXX_NODISCARD_ATTRIBUTE _LIBCUDACXX_DEVICE
+inline barrier<thread_scope_block>::arrival_token arrive_tx(
```

Comment:
If `__CUDA_MINIMUM_ARCH__ < 900`, this function should be a wrapper that `static_asserts` on use with a clear error message, to avoid requiring users to have to re-implement these checks.

Reply:
I agree that would be the most user-friendly way to do it. I have now changed the code to ifdef the function out when cuda minimum arch is pre-sm90. This is a very C++-like user-hostile way of solving the problem, but since the function may or may be a function template this is the only way in light of https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2022/p2593r0.html